src/Interpreters/Streaming/Aggregator.cpp

```diff
@@ -282,8 +389,21 @@ Aggregator::Aggregator(const Params & params_) : params(params_),  log(&Poco::Lo
     total_size_of_aggregate_states = 0;
     all_aggregates_has_trivial_destructor = true;
 
+    if (params.tracking_changes)
+    {
+        total_size_of_aggregate_states = sizeof(RetractedDataEx);
+        align_aggregate_states = alignof(RetractedDataEx);
+        expanded_data_type = ExpandedDataType::UpdatedWithRetracted;
+    }
+    else if (params.tracking_updated)
+    {
+        total_size_of_aggregate_states = sizeof(UpdatedDataEx);
+        align_aggregate_states = alignof(UpdatedDataEx);
+        expanded_data_type = ExpandedDataType::Updated;
+    }
+
     // aggregate_states will be aligned as below:
-    // |<-- state_1 -->|<-- pad_1 -->|<-- state_2 -->|<-- pad_2 -->| .....
+    // |<-- [ExpandedDataEx] -->||<-- state_1 -->|<-- pad_1 -->|<-- state_2 -->|<-- pad_2 -->| .....
```

Comment:
Does it have to be at the head, can we do it at the tail ? Just wondering the pros and cons

Reply:
Both are good, but if we do it at the tail ,  need extra variable to save the offset